drivers/led_strip/ws2812b_sw.c

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright (c) 2018 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <led_strip.h>
+
+#include <string.h>
+
+#define SYS_LOG_LEVEL CONFIG_SYS_LOG_LED_STRIP_LEVEL
+#include <logging/sys_log.h>
+
+#include <zephyr.h>
+#include <board.h>
+#include <gpio.h>
+#include <device.h>
+#include <clock_control.h>
+
+#define BLOCKING ((void *)1)
+
+static int send_buf(u8_t *buf, size_t len)
+{
+	/* Address of OUTSET. OUTCLR is OUTSET + 4 */
+	volatile u32_t *base = (u32_t *)(NRF_GPIO_BASE + 0x508);
```

Comment:
This is very NRF5 specific, do you want to conditional compile? or have depends on?

Thinking aloud, @dbkinder @SebastianBoe is there a build target that can list features/drivers and corresponding supported SoC/Board in Zephyr?
Example, make help -DFEATURES  or something

Reply:
The Kconfig entry has a `depends on CPU_CORTEX_M0`. Is that sufficient? I could add a pre-processor `#error` to the code as well in case the architecture isn't supported, however that'd be a bit redundant with the Kconfig dependency.

Reply:
Ah, right, CORTEX_M0 doesn't mean it's nRF, of course. I'll fix this, i.e. change the dependency to be for nRF51 for now.

Reply:
For the record, here's an example of multi-arch #ifdefs that another implementation uses:

https://github.com/adafruit/Adafruit_NeoPixel/blob/master/Adafruit_NeoPixel.cpp

However, I hope we can do something a bit cleaner than that.

Reply:
Could you have separate assembly files please, when you go for many variants. This inline asm in c is hard on eyes!